connect_ext_ppr/service.py

```diff
@@ -90,3 +97,136 @@ def update_product(data, config, logger):
+                state=Configuration.STATE.active,
+            ).first()
+        )
+        if not active_configuration:
+            raise ExtensionHttpError.EXT_006(
+                format_kwargs={'deployment_id': deployment.id},
+            )
+        config_kwargs.update({'configuration': active_configuration.id})
+        previous_ppr = (
+            db.query(PPRVersion)
+            .filter_by(
+                deployment=deployment.id,
+                status=PPRVersion.STATUS.ready,
+            )
+            .order_by(desc(PPRVersion.version))
+            .first()
+        )
+        data = None
+        config_json = get_configuration_from_media(
+            client, deployment.account_id, deployment.id, active_configuration.file,
+        )
+        product_info = (
+            f"(product_id={deployment.product_id}, "
+            f"product_version={deployment.product.version})"
+        )
+        print(previous_ppr)
+        if previous_ppr:
+            logger.info(
+                f"Start creation of PPR version {product_info}"
+                f" based on previous PPR file: '{previous_ppr.id}'.",
+            )
+            data = get_ppr_from_media(
+                client, deployment.account_id, deployment.id, previous_ppr.file,
+            )
+        else:
+            logger.info(
+                f"Start creation of PPR version {product_info}"
+                f" based on previous product information.",
+            )
+        items = list(get_product_items(client, deployment.product.id))
+
+        file, writer, wb = get_base_workbook(data)
+        ws_list, summary = process_ppr(wb, deployment.product, config_json, items)
+
+        for ws in ws_list:
+            ws.to_excel(writer, ws.name, index=False)
+
+        file_obj = open(file.name, 'rb')
+        file_obj.seek(0, os.SEEK_END)
+        file_size = file_obj.tell()
+        file_obj.seek(0)
+        writer.book.save(file_obj.name)
+
+        file_name = PPR_FILE_NAME.format(
+            product_id=deployment.product_id,
+            version=new_version,
+            timestamp=datetime.utcnow().strftime("%s"),
+        )
+        media_file = create_ppr_to_media(
+            client, deployment.account_id, deployment.id, file_name, file_obj.read(), file_size,
+        )
+        file.close()
+        file_data = FileSchema(
+            id=media_file['id'],
+            name=file_name,
+            location=media_file['file'],
+            size=file_size,
+            mime_type=File.MIME_TYPE.application_vnd_ms_xslx,
+        )
+    else:
+        summary = {}
+        data = get_ppr_from_media(client, deployment.account_id, deployment.id, file_data.id)
+        dict_file = workbook_to_dict(pd.ExcelFile(data))
+        errors = validate_ppr_schema(dict_file) or []
+        if errors:
+            summary.update({'errors': errors})
+            status = PPRVersion.STATUS.failed
+    try:
+        file_instance = File(
```

Comment:
I propose to leave this logic related to models in webapp.py for uniformity

Reply:
I agree on reduce the create function because is really big, but all related to file management, creating object is more related to the service layer than the webapp.